downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -204,7 +204,7 @@ abstract class VersionControlSystem {
      *
      * @return An object describing the downloaded working tree.
      *
-     * @throws DownloadException In case the download failed.
+     * @throws DownloadException in case the download failed.
```

Comment:
What do you think of renaming `DownloadException` to `VcsException` and using it for both cases, instead of throwing custom `DownloadException`s in one case and standard `IOException`s in the other?

Reply:
I don't think that renaming exceptions helps here. The underlying issue is that VCS implementations might use a CLI (which throws IOExceptions) or a library (which throws its custom exception). IMO the individual VCS implementations should abstract away this difference and only throw one exception to the outside / caller. The pragmatic choice it to align on IOException at this level, because then any existing IOExceptions do not have to wrapped and re-thrown. On the caller-level, i.e. the `download` function, we should then align on an ORT-specific exception, and `DownloadException` still fits here name-wise WRT the function name.

Reply:
I don't really see the difference between the `download` function and the others from a user perspective, but this does not need to be change now anyway.

Reply:
The difference is that `initWorkingTree()` / `updateWorkingTree()` are low-level functions that are not supposed to be called by the user directly usually; they are meant to be used by the default implementation of the high-level `download()` only. But yes, we can revisit this once the VCS / Subversion refactoring is finished.